index.js

```diff
@@ -79,17 +81,28 @@ function formatWithCursor(text, opts, addAlignmentSize) {
   const toStringResult = printDocToString(doc, opts);
   const str = toStringResult.formatted;
   const cursorOffsetResult = toStringResult.cursor;
+  const cursorNodeEnd = toStringResult.cursorNodeEnd;
   ensureAllCommentsPrinted(astComments);
   // Remove extra leading indentation as well as the added indentation after last newline
   if (addAlignmentSize > 0) {
     return { formatted: str.trim() + opts.newLine };
   }
 
   if (cursorOffset !== undefined) {
-    return {
-      formatted: str,
-      cursorOffset: cursorOffsetResult + cursorOffset
-    };
+    if (cursorOffsetResult + cursorOffset < cursorNodeEnd) {
+      return {
+        formatted: str,
+        cursorOffset: cursorOffsetResult + cursorOffset
+      };
+    } else {
+      return {
```

Comment:
You just need this `return` right? Math.max should remove the need for the condition right?